### ISSUE TITLE
Bug in variable declaration

### ISSUE THREAD

**Issue by anouard24** [chermehdi/comet#4]
I tried this:
```go
var a = 7
var a = "Test"
```
and it works, it erase old value of `a` instead of returning a `CometError`.

We have to fix this and I think, We must implement something to not allowing variable shadowing either, it's a source of bugs.

**Comment by chermehdi** [comment]
Well this is a language design issue, this is possible as well in some languages, it could be that i won't allow it in comet, but until I have a formal specification for the language this issue can still wait as I might decide to keep this in the language.

**Comment by anouard24** [comment]
Yeah, but if it stay like that, we can drop the word `var`, and use a declaration design like Python or PHP

**Comment by zHaytam** [comment]
Personally, I'd rather see:
```
var a = 7
a = "Test"
```

Instead of having the possibility to create the same variable in the same scope (or child).  
It's possible in JS, but we all know JS isn't the greatest...

**Comment by chermehdi** [comment]
@zHaytam sorry for not seeing the comment earlier got lost in my email.

For you suggestion above, it's a valid point of view and I agree with it to some extent, but as I said earlier having it this way  or not
is not a decision we need to make right now in the language, because scoping is not easy as we need to define scoping rules for other language constructs (functions, structs ...) and as long as this is not defined I don't think this hurts / or helps, i.e it's not a priority for the time being and doing the fix to make the evaluator report an error if this happens is just a couple of lines, so in the future if this aligns with the scoping rules of the other constructs making the fix is not gonna be an issue.
